src/ext/php5/engine_hooks.c

```diff
@@ -395,27 +734,8 @@ void ddtrace_execute_internal(zend_execute_data *execute_data, zend_fcall_info *
 #endif
     }
 
-    zend_op *opline = EX(opline);
-    ddtrace_span_fci *span_fci = ecalloc(1, sizeof(*span_fci));
-    span_fci->execute_data = execute_data;
-    span_fci->dispatch = dispatch;
-    span_fci->fbc = fbc;
-    span_fci->This = This;
-    span_fci->called_scope = fbc->common.scope ? EG(called_scope) : NULL;
-    span_fci->arguments = EX(function_state).arguments;
-
-    ddtrace_open_span(span_fci TSRMLS_CC);
-
-    _dd_prev_execute_internal(execute_data, fci, return_value_used TSRMLS_CC);
-
-    if (span_fci == DDTRACE_G(open_spans_top)) {
-        _dd_execute_end_span(span_fci, *fci->retval_ptr_ptr, opline TSRMLS_CC);
-    } else {
-        if (get_dd_trace_debug()) {
-            const char *fname = Z_STRVAL(dispatch->function_name);
-            ddtrace_log_errf("Cannot run tracing closure for %s(); spans out of sync", fname);
-        }
-    }
+    uint16_t dispatch_type = dispatch->options & 3u;
```

Comment:
The magic number is a bit easier to understand with this temp var than it is in the PHP 7 code. What are your thoughts on making a macro for this? :) *ducks*

Reply:
Macro made :)